src/jvmMain/kotlin/io/github/moreirasantos/pgkn/PostgresDriver.kt

```diff
@@ -36,21 +64,25 @@ suspend fun PostgresDriver(
             .option(USER, user)
             .option(PASSWORD, password)
             .option(DATABASE, database)
+            .option(INITIAL_SIZE, poolSize)
             .option(MAX_SIZE, poolSize)
             .build()
-    )
-
-    val connection = connectionFactory.create().awaitFirst()
-
-    return PostgresDriverPool(connection = connection)
-}
+    ).apply {
+        // https://github.com/r2dbc/r2dbc-pool
+        // ConnectionFactories.get(..) creates a [ConnectionPool] wrapping an underlying [ConnectionFactory].
+        // So, we can warm up the pool here (blocking).
+        (this as? ConnectionPool)?.let { runBlocking { warmup().awaitFirst() } }
```

Comment:
We shouldn't use `runBlocking` in a library - all IO should go through `suspend` in some shape or form.

Side note: these any IO using libpq will be blocking, because libpq itself is blocking, but the user of this library should, in the future, be able to pass the coroutine.IO context for these cases  - like https://github.com/JetBrains/Exposed does. In the future though, this library shall use a pure-kotlin SSL/TLS implementation to connect to DB, making it fully non-blocking.

Reply:
You are right on that.
I can remove the `warmup` from the "constructor" and I can provide a `warmup` method through the `PostgresDriver` interface.
What do you think about that?

Reply:
Yep, something like an Actor or similar? Of course, it has to make sense with kotlin/native as well. Maybe we can warmup after instantiating the Pool, inside the `PostgresDriver` method? I think we woudn't need to change the API like that (other than converting it to `suspend`)

Reply:
I think it would be better to just provide a `warmup` method through the `PostgresDriver` interface, thus let the client to decide if wants to warmup or not.